cpp/tests/gis/perimeter_test.cu

```diff
@@ -0,0 +1,238 @@
+    void validation(std::vector<T> poly_concat_lats[], std::vector<T> poly_concat_lons[], int n_polygons)
+    {
+        size_t min_polygon_size_accepted = 4;
+
+        for (int i = 0; i < n_polygons; ++i)
+        {
+            EXPECT_LE( min_polygon_size_accepted, poly_concat_lats[i].size() ) << "TEST: Unbuilt polygon";
+            EXPECT_LE( min_polygon_size_accepted, poly_concat_lons[i].size() ) << "TEST: Unbuilt polygon";
+
+            // Polygon must be closed
+            size_t size_polygon_i = poly_concat_lats[i].size();
+            EXPECT_EQ( poly_concat_lats[i][0], poly_concat_lats[i][size_polygon_i - 1] ) << "TEST: Latitudes of polygon must be closed.";
+            EXPECT_EQ( poly_concat_lons[i][0], poly_concat_lons[i][size_polygon_i - 1] ) << "TEST: Longitudes of polygon must be closed.";
+        }
+    }
+
+    // Initialize random polygons with a specific range
+    void initialize_random( std::vector<T> poly_concat_lats[], std::vector<T>  poly_concat_lons[], 
+                            const int & n_polygons, const int & size_polygon, const T & min_value_lats,
+                            const T & max_value_lats, const T & min_value_lons, const T & max_value_lons,
+                            bool print = false )
+    {
+        for (int i = 0; i < n_polygons; ++i)
+        {
+            poly_concat_lats[i].resize(size_polygon);
+            poly_concat_lons[i].resize(size_polygon);
+        }
+
+        for (int i = 0; i < n_polygons; ++i)
+        {
+            for (int j = 0; j < size_polygon; ++j)
+            {
+                RandomValues<T>random_lat(min_value_lats, max_value_lats);
+                poly_concat_lats[i][j] = random_lat();
+                RandomValues<T>random_lon(min_value_lons, max_value_lons);
+                poly_concat_lons[i][j] = random_lon();
+            }
+        }
+
+        // First and last points of each polygon must be equal
+        for (int i = 0; i < n_polygons; ++i)
+        {
+            poly_concat_lats[i][size_polygon - 1] = poly_concat_lats[i][0];
+            poly_concat_lons[i][size_polygon - 1] = poly_concat_lons[i][0];
+        }
+    }
+
+    std::vector<T> compute_reference_perimeter(std::vector<T> poly_concat_lats[], std::vector<T> poly_concat_lons[], int n_polygons)
+    {
+        validation(poly_concat_lats, poly_concat_lons, n_polygons);
+        std::vector<T> host_perimeter(n_polygons);
+        
+        for (int poly_i = 0; poly_i < n_polygons; ++poly_i)
+        {
+            int size_poly_i = poly_concat_lats[poly_i].size();
+            double perimeter = 0.0;
+            for (int side_i = 0; side_i < size_poly_i - 1; ++side_i)
+            {
+                T lat_variation = (poly_concat_lats[poly_i][side_i + 1] - poly_concat_lats[poly_i][side_i]) * DEG_TO_RAD;
+                T lon_variation = (poly_concat_lons[poly_i][side_i + 1] - poly_concat_lons[poly_i][side_i]) * DEG_TO_RAD;
+                T partial_1 = sin(lat_variation * 0.5);
```

Comment:
Put the haversine computation in a function.

Reply:
Ok, done!